src/logging/writers/sqlite/sqlite.bif

```diff
@@ -7,3 +7,23 @@ const set_separator: string;
 const empty_field: string;
 const unset_field: string;
 
+enum SQLiteSynchronous %{
+	SQLITE_SYNCHRONOUS_DEFAULT,
+	SQLITE_SYNCHRONOUS_OFF,
+	SQLITE_SYNCHRONOUS_NORMAL,
+	SQLITE_SYNCHRONOUS_FULL,
+	SQLITE_SYNCHRONOUS_EXTRA,
+%}
+
+enum SQLiteJournalMode %{
+	SQLITE_JOURNAL_MODE_DEFAULT,
+	SQLITE_JOURNAL_MODE_DELETE,
+	SQLITE_JOURNAL_MODE_TRUNCATE,
+	SQLITE_JOURNAL_MODE_PERSIST,
+	SQLITE_JOURNAL_MODE_MEMORY,
+	SQLITE_JOURNAL_MODE_WAL,
+	SQLITE_JOURNAL_MODE_OFF,
+%}
```

Comment:
We replicate those enums in 3 places. It seems using the one here is just enough (with the drawback of not having  docs explicitly in sqlite.zeek).

IMO at least the ones in SQLite.h could be removed in favor of the BifEnum generated ones? Here's a hunk that seems to work out okay:
```
diff --git a/src/logging/writers/sqlite/SQLite.cc b/src/logging/writers/sqlite/SQLite.cc
index 12dc7ba3cc..5186c14a5c 100644
--- a/src/logging/writers/sqlite/SQLite.cc
+++ b/src/logging/writers/sqlite/SQLite.cc
@@ -134,13 +134,17 @@ bool SQLite::DoInit(const WriterInfo& info, int arg_num_fields, const Field* con
     char* errorMsg = nullptr;
     int res;
     switch ( synchronous ) {
-        case SQLITE_SYNCHRONOUS_DEFAULT: res = SQLITE_OK; break;
-        case SQLITE_SYNCHRONOUS_OFF: res = sqlite3_exec(db, "PRAGMA synchronous=OFF;", NULL, NULL, &errorMsg); break;
-        case SQLITE_SYNCHRONOUS_NORMAL:
+        case BifEnum::LogSQLite::SQLiteSynchronous::SQLITE_SYNCHRONOUS_DEFAULT: res = SQLITE_OK; break;
+        case BifEnum::LogSQLite::SQLiteSynchronous::SQLITE_SYNCHRONOUS_OFF:
+            res = sqlite3_exec(db, "PRAGMA synchronous=OFF;", NULL, NULL, &errorMsg);
+            break;
+        case BifEnum::LogSQLite::SQLiteSynchronous::SQLITE_SYNCHRONOUS_NORMAL:
             res = sqlite3_exec(db, "PRAGMA synchronous=NORMAL;", NULL, NULL, &errorMsg);
             break;
-        case SQLITE_SYNCHRONOUS_FULL: res = sqlite3_exec(db, "PRAGMA synchronous=FULL;", NULL, NULL, &errorMsg); break;
-        case SQLITE_SYNCHRONOUS_EXTRA:
+        case BifEnum::LogSQLite::SQLiteSynchronous::SQLITE_SYNCHRONOUS_FULL:
+            res = sqlite3_exec(db, "PRAGMA synchronous=FULL;", NULL, NULL, &errorMsg);
+            break;
+        case BifEnum::LogSQLite::SQLiteSynchronous::SQLITE_SYNCHRONOUS_EXTRA:
             res = sqlite3_exec(db, "PRAGMA synchronous=EXTRA;", NULL, NULL, &errorMsg);
             break;
         default: Error("Invalid LogSQLite::synchronous enum"); return false;
@@ -153,21 +157,25 @@ bool SQLite::DoInit(const WriterInfo& info, int arg_num_fields, const Field* con
     }
 
     switch ( journal_mode ) {
-        case SQLITE_JOURNAL_MODE_DEFAULT: res = SQLITE_OK; break;
-        case SQLITE_JOURNAL_MODE_DELETE:
+        case BifEnum::LogSQLite::SQLiteJournalMode::SQLITE_JOURNAL_MODE_DEFAULT: res = SQLITE_OK; break;
+        case BifEnum::LogSQLite::SQLiteJournalMode::SQLITE_JOURNAL_MODE_DELETE:
             res = sqlite3_exec(db, "PRAGMA journal_mode=DELETE;", NULL, NULL, &errorMsg);
             break;
-        case SQLITE_JOURNAL_MODE_TRUNCATE:
+        case BifEnum::LogSQLite::SQLiteJournalMode::SQLITE_JOURNAL_MODE_TRUNCATE:
             res = sqlite3_exec(db, "PRAGMA journal_mode=TRUNCATE;", NULL, NULL, &errorMsg);
             break;
-        case SQLITE_JOURNAL_MODE_PERSIST:
+        case BifEnum::LogSQLite::SQLiteJournalMode::SQLITE_JOURNAL_MODE_PERSIST:
             res = sqlite3_exec(db, "PRAGMA journal_mode=PERSIST;", NULL, NULL, &errorMsg);
             break;
-        case SQLITE_JOURNAL_MODE_MEMORY:
+        case BifEnum::LogSQLite::SQLiteJournalMode::SQLITE_JOURNAL_MODE_MEMORY:
             res = sqlite3_exec(db, "PRAGMA journal_mode=MEMORY;", NULL, NULL, &errorMsg);
             break;
-        case SQLITE_JOURNAL_MODE_WAL: res = sqlite3_exec(db, "PRAGMA journal_mode=WAL;", NULL, NULL, &errorMsg); break;
-        case SQLITE_JOURNAL_MODE_OFF: res = sqlite3_exec(db, "PRAGMA journal_mode=OFF;", NULL, NULL, &errorMsg); break;
+        case BifEnum::LogSQLite::SQLiteJournalMode::SQLITE_JOURNAL_MODE_WAL:
+            res = sqlite3_exec(db, "PRAGMA journal_mode=WAL;", NULL, NULL, &errorMsg);
+            break;
+        case BifEnum::LogSQLite::SQLiteJournalMode::SQLITE_JOURNAL_MODE_OFF:
+            res = sqlite3_exec(db, "PRAGMA journal_mode=OFF;", NULL, NULL, &errorMsg);
+            break;
         default: Error("Invalid LogSQLite::journal_mode enum"); return false;
     }
 
diff --git a/src/logging/writers/sqlite/SQLite.h b/src/logging/writers/sqlite/SQLite.h
index be488cecbf..5fd20bf45f 100644
--- a/src/logging/writers/sqlite/SQLite.h
+++ b/src/logging/writers/sqlite/SQLite.h
@@ -48,24 +48,6 @@ private:
     int64_t synchronous;
     int64_t journal_mode;
 
-    enum SQLiteSynchronous {
-        SQLITE_SYNCHRONOUS_DEFAULT,
-        SQLITE_SYNCHRONOUS_OFF,
-        SQLITE_SYNCHRONOUS_NORMAL,
-        SQLITE_SYNCHRONOUS_FULL,
-        SQLITE_SYNCHRONOUS_EXTRA,
-    };
-
-    enum SQLiteJournalMode {
-        SQLITE_JOURNAL_MODE_DEFAULT,
-        SQLITE_JOURNAL_MODE_DELETE,
-        SQLITE_JOURNAL_MODE_TRUNCATE,
-        SQLITE_JOURNAL_MODE_PERSIST,
-        SQLITE_JOURNAL_MODE_MEMORY,
-        SQLITE_JOURNAL_MODE_WAL,
-        SQLITE_JOURNAL_MODE_OFF,
-    };
-
     threading::formatter::Ascii* io;
 };


```

Reply:
Thank you - that patch definitely makes sense.

I am slightly in favor of keeping the one in the zeek file - to make finding them easier for people looking at the docs.